vmcnet/models/construct.py

```diff
@@ -507,10 +515,13 @@ def backflow_based_equivariance(x: ArrayList) -> Array:
 def get_residual_blocks_for_ferminet_backflow(
     spin_split: ParticleSplit,
     ndense_list: List[Tuple[int, ...]],
+    num_heads: int,
```

Comment:
Hm could we make this an optional parameter after use_transformer, and call it transformer_num_heads?

Similar to how we have two_electron_skip and two_electron_skip_scale